src/view/reviewManager.ts

```diff
@@ -1026,13 +1026,83 @@ export class ReviewManager implements vscode.DecorationProvider {
 		await this.validateState();
 	}
 
+	public async publishBranch(): Promise<Branch> {
+		const HEAD = this._repository.state.HEAD;
+		const githubRemotes = this._prManager.getGitHubRemotes();
+		const picks: RemoteQuickPickItem[] = githubRemotes.map(remote => new RemoteQuickPickItem(remote));
+		const selected: RemoteQuickPickItem = await vscode.window.showQuickPick<RemoteQuickPickItem>(picks, {
+			ignoreFocusOut: true,
+			placeHolder: `Pick a remote to publish the branch ${HEAD.name} to:`
+		});
+
+		if (!selected) {
+			return;
+		}
+
+		let upstreamBranchName = await vscode.window.showInputBox({
```

Comment:
I'd suggest dropping this prompt and just pushing the branch with the same name. Pushing with a different name is pretty uncommon IME (but maybe there's a use case I'm not aware of?). And anyway, if the user wants that, they can push it themselves.